torchgeo/samplers/single.py

```diff
@@ -44,18 +113,103 @@ def __init__(self, dataset: GeoDataset, roi: BoundingBox | None = None) -> None:
 
         self.res = dataset.res
         self.roi = roi
+        self.dataset = dataset
 
     @abc.abstractmethod
+    def get_chips(self, *args: Any, **kwargs: Any) -> GeoDataFrame:
```

Comment:
Body of the function missing or is this simply a placeholder?

Reply:
It is an abstract method, so all classes implementing GeoSampler should define a function called get_chips. Python automatically raises an error if your class does not have a get_chips function